test/units/galaxy/test_token.py

```diff
@@ -96,3 +190,51 @@ def test_token_from_file_missing(b_token_file):
 @pytest.mark.parametrize('b_token_file', ['file'], indirect=True)
 def test_token_none(b_token_file):
     assert GalaxyToken(token=NoTokenSentinel).get() is None
+
+
+def test_parse_bearer_token(monkeypatch):
+    monkeypatch.setattr(token_file, 'REQUIRED_JWT_HEADER_KEYS', set(['alg']))
+    monkeypatch.setattr(token_file, 'REQUIRED_JWT_HEADER_VALUES', {'alg': ['RS256']})
+
+    # result from https://www.rfc-editor.org/rfc/rfc7515#appendix-A.2.1 to test validating A.2.2
+    token_info = (
+        "eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJqb2UiLA0KICJleHAiOjEzMDA4MTkzODAsDQogImh0"
+        "dHA6Ly9leGFtcGxlLmNvbS9pc19yb290Ijp0cnVlfQ.cC4hiUPoj9Eetdgtv3hF80EGrhuB__"
+        "dzERat0XF9g2VtQgr9PJbu3XOiZj5RZmh7AAuHIm4Bh-0Qc_lF5YKt_O8W2Fp5jujGbds9uJd"
+        "bF9CUAr7t1dnZcAcQjbKBYNX4BAynRFdiuB--f_nZLgrnbyTyWzO75vRK5h6xBArLIARNPvkS"
+        "jtQBMHlb1L07Qe7K0GarZRmB_eSN9383LcOLn6_dO--xi12jzDwusC-eOkHWEsqtFZESc6BfI"
+        "7noOPqvhJ1phCnvWh6IeYI2w9QOYEUipUTI8np6LbgGY9Fs98rqVt5AXLIhWkWywlVmtVrBp0"
+        "igcN_IoypGlUPQGe77Rw"
+    )
+
+    e = "AQAB"
+    n = (
+        "ofgWCuLjybRlzo0tZWJjNiuSfb4p4fAkd_wWJcyQoTbji9k0l8W26mPddxHmfHQp-Vaw-4qPC"
+        "JrcS2mJPMEzP1Pt0Bm4d4QlL-yRT-SFd2lZS-pCgNMsD1W_YpRPEwOWvG6b32690r2jZ47soM"
+        "Zo9wGzjb_7OMg0LOL-bSf63kpaSHSXndS5z5rexMdbBYUsLA9e-KXBdQOS-UTo7WTBEMa2R2C"
+        "apHg665xsmtdVMTBQY4uDZlxvb3qCo5ZwKh9kG4LT6_I5IhlJH7aGhyxXFvUK-DWNmoudF8NA"
+        "co9_h9iaGNj8q2ethFkMLs91kzk2PAcDTW9gb54h4FRWyuXpoQ"
+    )
+    public_numbers = token_file.PublicNumbers(n=n, e=e)
+    token_file.parse_bearer_token(token_info, public_numbers=public_numbers)
+
+    with pytest.raises(token_file.InvalidSignature):
+        token_file.parse_bearer_token(token_info + 'wrongsig', public_numbers=public_numbers)
+
+
+@pytest.mark.parametrize('exp_delta', [1, -15, 15])
+def test_keycloak_expiration(exp_delta, monkeypatch, private_key, public_key_numbers):
+    expected_expired_result = exp_delta <= 1
+
+    exp_timestamp = int((datetime.now(tz=timezone.utc) + timedelta(minutes=exp_delta)).timestamp())
+    signed_payload = create_token_payload(private_key, exp_timestamp)
+
+    token = KeycloakToken(access_token='', auth_url='')
+    token.jwks = [{'kid': 'key_id', 'e': public_key_numbers['e'], 'n': public_key_numbers['n']}]
+
+    mock_response = json.dumps({"access_token": "token", "id_token": signed_payload})
+    with StringIO(mock_response) as response:
+        monkeypatch.setattr(token_file, 'open_url', MagicMock(return_value=response))
```

Comment:
pro tip: we have `pytest-mock` integrated, meaning that we can inject the `mocker` fixture and access this through `mocker.MagicMock` instead of relying on the imports external to the pytest machinery.

Reply:
Ok, I removed the existing import and updated the whole file.